lib/connection-manager.js

```diff
@@ -65,21 +27,44 @@ module.exports.connectLogServer = function(config, programName) {
     };
     var proxy = getProxyFromURI(url.parse(logurl));
     if (proxy) options.agent = new HttpsProxyAgent(url.parse(proxy));
-
-    var firstrun = true;
     debug(logurl);
-    var socket = new WebSocket(logurl, options);
-    linebuffer.setSocket(socket);
-    debug(socket.options);
-    return new Promise(resolve => {
-        debug('connecting to websocket');
-        socket.on('open', function() {
-            log('Connected to ' + config.LOGDNA_LOGHOST + ':' + config.LOGDNA_LOGPORT + ' (' + socket._socket.remoteAddress + ')' + (config.LOGDNA_LOGSSL ? ' (SSL)' : '') + (proxy ? ' via ' + proxy : ''));
-
-            if (firstrun) {
-                firstrun = false;
-
-                // start streaming logs from logdir(s) on startup
+    const socket = new WebSocket(logurl, options);
+    const req = linebuffer.setConnection(logurl, options);
+    return new Promise((resolve) => {
+        async.retry({
+            times: 5
+            , interval: 100
+            , errorFilter: (err) => {
+                return [
+                    'ETIMEDOUT'
+                    , 'ECONNRESET'
+                    , 'EHOSTUNREACH'
+                    , 'ETIMEDOUT'
```

Comment:
You listed this twice?

Reply:
No, but had to make it as a constant since called in separate places.